go/ephemeral/common.go

```diff
@@ -44,3 +46,45 @@ func getCurrentUserUV(ctx context.Context, g *libkb.GlobalContext) (ret keybase1
 	})
 	return ret, err
 }
+
+// Map generations to their creation time
+type keyExpiryMap map[keybase1.EkGeneration]keybase1.Time
+
+// Keys expire after `KeyLifetimeSecs` unless there has been a gap in their
+// generation. If there has been a gap, a key can be re-used for up to
+// `KeyLifetimeSecs` until it is considered stale. To determine expiration, we
+// look at all of the current keys and account for any gaps since we don't want
+// to expire a key if it is still used to encrypt a different key. This only
+// applies to deviceEKs or userEKs since they can have a dependency above them.
+// A teamEK expires after `KeyLifetimeSecs` without exception, so it doesn't
+// call this.
```

Comment:
I suppose in our design there's always a gap between keys. It's just that that gap is only 24hrs if the device is behaving well. It might make sense to word the explanation from that perspective.